integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/plan/column/ColumnLevelLineageBuilder.java

```diff
@@ -163,19 +187,42 @@ public ColumnLineageDatasetFacetFields build() {
   }
 
   private List<OpenLineage.ColumnLineageDatasetFacetFieldsAdditionalInputFields> facetInputFields(
-      List<Pair<DatasetIdentifier, String>> inputFields) {
-    return inputFields.stream()
+      List<TransformedInput> inputFields, List<TransformedInput> datasetDependencyInputs) {
+
+    Stream<TransformedInput> concat =
+        Streams.concat(inputFields.stream(), datasetDependencyInputs.stream());
+    Map<Input, List<TransformedInput>> collect =
+        concat.collect(Collectors.groupingBy(TransformedInput::getInput, Collectors.toList()));
+
+    return collect.entrySet().stream()
         .map(
             field ->
                 new OpenLineage.ColumnLineageDatasetFacetFieldsAdditionalInputFieldsBuilder()
-                    .namespace(field.getLeft().getNamespace())
-                    .name(field.getLeft().getName())
-                    .field(field.getRight())
+                    .namespace(field.getKey().getDatasetIdentifier().getNamespace())
+                    .name(field.getKey().getDatasetIdentifier().getName())
+                    .field(field.getKey().getName())
+                    .transformations(
+                        field.getValue().stream()
+                            .map(TransformedInput::getTransformationInfo)
+                            .map(ColumnLevelLineageBuilder::inputFieldTransformations)
+                            .collect(Collectors.toList()))
                     .build())
         .collect(Collectors.toList());
   }
 
-  List<Pair<DatasetIdentifier, String>> getInputsUsedFor(String outputName) {
+  @NotNull
+  private static OpenLineage.ColumnLineageDatasetFacetFieldsAdditionalInputFieldsTransformations
```

Comment:
wouldn't this fit better into TranformationTypeInfo class as a `toInputFieldsTransformations`?